components/tidb_query_expr/src/impl_time.rs

```diff
@@ -443,6 +465,55 @@ pub fn sub_datetime_and_string(
     Ok(Some(res))
 }
 
+#[rpn_fn(capture = [ctx])]
+#[inline]
+pub fn sub_date_and_string(
```

Comment:
Seems that `sub_date_and_string` should return a string. Would you please double-check whether I'm correct? Refer to https://github.com/pingcap/tidb/blob/master/expression/builtin_time.go#L6590 for more information.

Reply:
Also https://github.com/tikv/tikv/blob/1a88f12ebf50064b992fc5efc7e7f55795210521/components/tidb_query_normal_expr/src/builtin_time.rs#L595

Reply:
ok, I will check them.